services/wireguard/endpoint/kernelspace/client.go

```diff
@@ -84,20 +93,47 @@ func (c *client) AddPeer(iface string, peer wg.PeerInfo) error {
 
 func (c *client) up(iface string, ipAddr net.IPNet) error {
 	if d, err := c.wgClient.Device(iface); err != nil || d.Name != iface {
-		if err := exec.Command("ip", "link", "add", "dev", iface, "type", "wireguard").Run(); err != nil {
+		if err := utils.SuExec("ip", "link", "add", "dev", iface, "type", "wireguard"); err != nil {
 			return err
 		}
 	}
 
-	if err := exec.Command("ip", "address", "replace", "dev", iface, ipAddr.String()).Run(); err != nil {
+	if err := utils.SuExec("ip", "address", "replace", "dev", iface, ipAddr.String()); err != nil {
 		return err
 	}
 
-	return exec.Command("ip", "link", "set", "dev", iface, "up").Run()
+	return utils.SuExec("ip", "link", "set", "dev", iface, "up")
+}
+
+func (c *client) excludeRoute(endpoint net.UDPAddr) error {
```

Comment:
Why UDPAddr is needed to excludeRoute - simple ip should be enough. UDP has nothing to do with routes

Reply:
Changed it.